### ISSUE TITLE
UA to block

### ISSUE THREAD

**Issue by mikeruss1** [CIDRAM/CIDRAM#440]
User agent: GuzzleHttp/7

it has a few UA variations - prob best to look for "Guzzle"

**Comment by mikeruss1** [comment]
another one - DnBCrawler-Analytics

**Comment by Maikuolan** [comment]
DnBCrawler-Analytics should now be blocked by the bot UAs module. Cheers. :-)

Not sure about blocking Guzzle. There are bots that use it, but there are also a number of legitimate crawlers and webtools that use it, too.

*Related: https://docs.guzzlephp.org/en/stable/*

**Comment by eurobank** [comment]
I think users of CIDRAM must drop bobuam module because of large ammount of wrong blocks (and i guess also missed blocks) and use Aux Rules for user agents block.

**Comment by eurobank** [comment]
Here is what i use:

![Screenshot](https://user-images.githubusercontent.com/4467158/225971513-9c849e76-6ec3-47ae-9052-6fa1401bae18.png)

**Comment by Maikuolan** [comment]
> I think users of CIDRAM must drop bobuam module because of large ammount of wrong blocks (and i guess also missed blocks also) and use Aux Rules for user agents block.

As things stand right now today, it *could* be argued either way, depending on the exact circumstances of the users in question, the needs of their websites, etc, and I think, there's definitely value in continuing to use BOBUAM today, for *some* users. That said though, whether or not they continue or stop using BOBUAM today, the prospect of dropping BOBUAM entirely is something that all users will likely eventually need to face at some point in the future due to various changes proposed by W3C and other organisations in recent years (starting with "client hints", but also covering other topics) which'll likely strongly affect how browsers present themselves, how useful and how reliable user agents are likely to be *at all* in the future and so on.

For now, I intend to continue supporting BOBUAM as best I can, and I'm not going to tell anyone right now that they need to drop it. But, that said, I do suspect these problems are likely to *increase* in frequency as time goes on, and probably become worse rather than better. Hopefully my suspicions are wrong, but.. they're probably not wrong.

In any case, if as many users as possible could become more familiar with auxiliary rules, and more users use auxiliary rules more often, I would be strongly in favour of that. Strictly speaking, whether signature files, modules, or auxiliary rules, it's equally possible for users to create their own (custom signature files, custom modules, custom auxiliary rules, etc), and possible for users to share those amongst each other, without restriction. It's even possible for users to create and provide their own upstream source for the updates page, if they so wanted (and if they had the time and desire to spend the effort necessarily to learn how to do that and maintain it). However, undeniably, it's much, *much* easier to do so with auxiliary rules. There's an interface at the front-end for creating, importing, and exporting auxiliary rules, so they can be easily created and easily shared; no PHP knowledge required at all. There's no such interface for signature files or modules though. The technical knowledge and skill required to create auxiliary rules doesn't extend much beyond knowing how to use the interface and knowing generally how auxiliary rules work, with a minimal learning curve. Despite having no interface, the technical knowledge and skill required to create custom signature files isn't too extreme, either. Modules, on the other hand, require someone to understand how to write PHP code, require them to understand the inner workings of CIDRAM (at least to the extent that it affects how CIDRAM interacts with modules), and would involve all usual the risk associated with writing one's own PHP code (or with using a stranger's code, in the event of using a module shared from another user). So, all in all, auxiliary rules are going to be easier (and safer) in general for CIDRAM users to work with than anything else.

The significance of that? If anything were to happen to those currently providing signature files and modules (which currently, is just me), the likelihood of that resulting in the complete stagnation or even, potentially, the total death of the CIDRAM project is much smaller, because users, upon finding new things which should be blocked and other such needed changes, could potentially share auxiliary rules with each other in order to keep things going. In other words, CIDRAM could potentially continue to grow and remain relevant in the absence of any maintainers due to the efforts of its users (i.e., a way to somewhat alleviate the effects of the "[bus factor](https://en.wikipedia.org/wiki/Bus_factor)").

Of course, I don't currently have any plans to quit, and don't expect anything bad to happen in the near future, but it's always nice to have mechanisms in place, just in case. ;-)

**Comment by eurobank** [comment]
@Maikuolan Sorry for the slight OT, but after i discovered Aux Rules (2 years later from the day i started to use CIDRAM), i think it is a pure gold (for me).

**ONLY the fact that i can block thousands of IPs (different CIDRs via 10 or 20 only ASNs, instead of manually adding IPs or CIDRs, make my life very easy.**

**Comment by mikeruss1** [comment]
personally I think a bot blocker is important. Most get stopped via "Cloud service", but a sizable minority dont. There are other solutions and one of the best is referred to in the update page for Bobuam, and was the original resource for the Bobuam content. I use aux rules as a temporary method to block or unblock. In the long term its not worth the effort of managing for every concerned website to manage bot rules. There are thousands.

**Comment by eurobank** [comment]
@mikeruss1 Cloud service (you mean the signatures i guess) are ALSO blocking several important IPs (aws for example). 

As for bots, daily new bots are appearing, so bobuam can't follow that.

**Comment by mikeruss1** [comment]
I was wondering if it were possible or useful, to split up the functions of Bobuam. There are plenty of sources for lists of bad bots, some of which get updated. If that capability was replaced by a signature file for users to maintain, would it be a better solution for Bobuam to continue the other functions of checking browser age, and UA format? Would it make your life easier?
